core/primitives/src/shard_layout.rs

```diff
@@ -194,7 +194,7 @@ impl ShardLayout {
     /// Returns error if `shard_id` is an invalid shard id in the current layout
     /// Panics if `self` has no parent shard layout
     pub fn get_parent_shard_id(&self, shard_id: ShardId) -> Result<ShardId, ShardLayoutError> {
-        if shard_id > self.num_shards() {
+        if !self.shard_ids().any(|id| id == shard_id) {
```

Comment:
nit: contains

Reply:
`shard_ids()` is returning an iterator here, not a vector.  AFAICT, there is no `contains` function on iterators.